src/test/codewhisperer/util/crossFileContextUtil.test.ts

```diff
@@ -0,0 +1,168 @@
+import * as path from 'path'
+import * as sinon from 'sinon'
+import * as semver from 'semver'
+import * as crossFile from '../../../codewhisperer/util/supplementalContext/crossFileContextUtil'
+import { createMockTextEditor } from '../testUtil'
+import { CodeWhispererUserGroupSettings } from '../../../codewhisperer/util/userGroupUtil'
+import { UserGroup } from '../../../codewhisperer/models/constants'
+import { assertTabCount, closeAllEditors, createTestWorkspaceFolder, toFile } from '../../testUtil'
+
+let tempFolder: string
+
+async function openATextEditorWithText(fileText: string, fileName: string, folder: string): Promise<vscode.TextEditor> {
+    const completeFilePath = path.join(folder, fileName)
+    toFile(fileText, completeFilePath)
+
+    const textDocument = await vscode.workspace.openTextDocument(completeFilePath)
+
+    return await vscode.window.showTextDocument(textDocument, { preview: false })
+}
+
+// VSCode tab APIs are available since 1.68.0
+function shouldRunTheTest(): boolean {
+    return (semver.valid(vscode.version) && semver.gte(vscode.version, '1.68.0')) as boolean
+}
+
+describe('crossFileContextUtil', function () {
+    const userGroupSettings = CodeWhispererUserGroupSettings.instance
+
+    const fakeCancellationToken: vscode.CancellationToken = {
+        isCancellationRequested: false,
+        onCancellationRequested: sinon.spy(),
+    }
+
+    let mockEditor: vscode.TextEditor
+
+    describe('non supported language should return undefined', function () {
+        it('c++', async function () {
+            mockEditor = createMockTextEditor('content', 'fileName', 'cpp')
+            const actual = await crossFile.fetchSupplementalContextForSrc(mockEditor, fakeCancellationToken)
+            assert.strictEqual(actual, undefined)
+        })
+
+        it('ruby', async function () {
+            mockEditor = createMockTextEditor('content', 'fileName', 'ruby')
+
+            const actual = await crossFile.fetchSupplementalContextForSrc(mockEditor, fakeCancellationToken)
+
+            assert.strictEqual(actual, undefined)
+        })
+    })
+
+    describe('partial support - control group', function () {
+        before(async function () {
+            this.timeout(60000)
+        })
+
+        beforeEach(async function () {
+            tempFolder = (await createTestWorkspaceFolder()).uri.fsPath
+        })
+
+        afterEach(async function () {
+            sinon.restore()
+            userGroupSettings.reset()
+            await closeAllEditors()
+        })
+
+        it('should be empty if userGroup is control', async function () {
+            if (!shouldRunTheTest) {
+                this.skip()
+            }
+
+            sinon.stub(userGroupSettings, 'userGroup').returns(UserGroup.Control)
+
+            // TODO: flaky, is there other approach to achive this? need this probably because it's an async test
+            while (userGroupSettings.userGroup !== UserGroup.Control) {}
```

Comment:
@justinmk3 @JadenSimon, naive q, I need this waiting probably because it's an async test, is there better approach to deal with it? thanks!